src/containers/Node/Node.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import {useRouteMatch} from 'react-router';
+import cn from 'bem-cn-lite';
+import {useDispatch, useSelector} from 'react-redux';
+import _ from 'lodash';
+
+import {Loader, Tabs} from '@yandex-cloud/uikit';
+import {Link} from 'react-router-dom';
+
+import {TABLETS, STORAGE, NODE_PAGES, OVERVIEW, STRUCTURE} from './NodePages';
+import Tablets from '../Tablets/Tablets';
+import Storage from '../Storage/Storage';
+import NodeOverview from './NodeOverview/NodeOverview';
+import NodeStructure from './NodeStructure/NodeStructure';
+
+import {getNodeInfo, resetNode} from '../../store/reducers/node';
+import routes, {CLUSTER_PAGES, createHref} from '../../routes';
+import {setHeader} from '../../store/reducers/header';
+import {AutoFetcher} from '../../utils/autofetcher';
+
+import './Node.scss';
+
+const b = cn('node');
+
+export const STORAGE_ROLE = 'Storage';
+
+const headerNodes = {
+    text: CLUSTER_PAGES.nodes.title,
+    link: createHref(routes.cluster, {activeTab: CLUSTER_PAGES.nodes.id}),
+};
+
+const autofetcher = new AutoFetcher();
+
+interface NodeProps {
+    additionalNodesInfo?: any;
+    className?: string;
+}
+
+function Node(props: NodeProps) {
+    const dispatch = useDispatch();
+
+    const {wasLoaded, loading, error} = useSelector((state: any) => state.node);
+
+    const {node, nodeHost} = useSelector((state: any) => {
+        const data = state.node.data;
+        const node = data.SystemStateInfo ? data.SystemStateInfo[0] : undefined;
+        return {node, nodeHost: node?.Host};
+    });
+
+    const match =
+        useRouteMatch<{id: string; activeTab: string}>(routes.node) ?? Object.create(null);
+
+    const {id: nodeId, activeTab} = match.params;
+
+    const {activeTabVerified, nodeTabs} = React.useMemo(() => {
+        const hasStorage = _.find(node?.Roles, (el) => el === STORAGE_ROLE);
+        let activeTabVerified = activeTab;
+        if (!hasStorage && activeTab === STORAGE) {
+            activeTabVerified = OVERVIEW;
+        }
+        const nodePages = hasStorage ? NODE_PAGES : NODE_PAGES.filter((el) => el.id !== STORAGE);
+
+        const nodeTabs = nodePages.map((page) => {
+            return {
+                ...page,
+                title: page.name,
+            };
+        });
+
+        return {activeTabVerified, nodeTabs};
+    }, [activeTab, node]);
+
+    const renderLoader = () => {
```

Comment:
Why can't this component be moved outside of <Node /> definition?

Reply:
Yes, it should be separate component. When I will work on other pages I'll refactor loaders.